assets/src/edit-story/components/library/panes/media/karma/mediaFetching.karma.js

```diff
@@ -26,36 +26,54 @@ import { waitFor } from '@testing-library/react';
 import { Fixture, MEDIA_PER_PAGE } from '../../../../../karma/fixture';
 import { ROOT_MARGIN } from '../mediaPane';
 
-describe('MediaPane fetching', () => {
-  let fixture;
+const flagValues = [false, true];
+flagValues.forEach((value) => {
+  describe('MediaPane fetching', () => {
+    let fixture;
 
-  beforeEach(async () => {
-    fixture = new Fixture();
-    await fixture.render();
-  });
+    beforeEach(async () => {
+      fixture = new Fixture();
+      fixture.setFlags({ rowBasedGallery: value });
+      await fixture.render();
+    });
 
-  afterEach(() => {
-    fixture.restore();
-  });
+    afterEach(() => {
+      fixture.restore();
+    });
 
-  it('should fetch 2nd page', async () => {
-    const mediaLibrary = fixture.querySelector('[data-testid="mediaLibrary"]');
-    let mediaElements = fixture.querySelectorAll('[data-testid=mediaElement]');
+    it(
+      'should fetch 2nd page - `rowBasedGallery` flag set to ' + value,
+      async () => {
+        let mediaElements;
+        await waitFor(() => {
```

Comment:
To make for a more succinct test, this could be hoisted to its own function:

const mediaElements = waitForMediaElements({ expectedCount: MEDIA_PER_PAGE});